### PR TITLE
Add tip to first uninstall the Python-only build before installing the Full build

### PR DESCRIPTION
**Pull request by clarng**
## Why are these changes needed?

It is not clear from the docs that one needs to first uninstall the setup in "Building Ray (Python Only)" if one wants to switch to "Building Ray on Linux & MacOS (full)". Because of the symlinks in Python-only setup one must first delete them and the pip package before doing pip install -e that is in the full build

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [] This PR is not tested :(

run make html and verify the content:

Tip

If you are only editing Python files, follow instructions for [Building Ray (Python Only)](https://github.com/ray-project/ray/compare/cl_fbd?expand=1#python-develop) to avoid long build times.

If you already followed the instructions in [Building Ray (Python Only)](https://github.com/ray-project/ray/compare/cl_fbd?expand=1#python-develop) and want to switch to the Full build in this section, you will need to first delete the symlinks and uninstall Ray.

### CODE REVIEW

File: doc/source/ray-contribute/development.rst
Comment (DmitriGekhtman) [inline doc/source/ray-contribute/development.rst:12]: Maybe could use a bit more detail on how to delete the symlinks. Is it easy to identify the symlinks that need to be deleted?

File: doc/source/ray-contribute/development.rst
Comment (clarng) [reply doc/source/ray-contribute/development.rst:12]: I hit this problem, and steps in the doc worked for me:

# To uninstall, delete the symlinks first.
rm -rf <package path>/site-packages/ray # Path will be in the output of `setup-dev.py`.
pip uninstall ray # or `pip install -U <wheel>`

**Review by scv119 — APPROVED** [review]

**Comment by scv119** [comment]
need code owner approval. cc @richard4912

**Review by richardliaw — APPROVED** [review]

**Review by DmitriGekhtman — APPROVED** [review]

**Pull request merged**
